### PR TITLE
[doc] document how to do multi-platform binary builds

### PR DESCRIPTION
**Pull request by justinabrahms**
Signed-off-by: Justin Abrahms <justin@abrah.ms>

fixes https://github.com/slsa-framework/slsa-github-generator/issues/1515

### CODE REVIEW

File: internal/builders/go/README.md [hunk 1]
```diff
@@ -144,6 +144,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
 | `{{ .Patch }}`       | `$(git describe --tags --always --dirty \| cut -d '.' -f3 \| cut -d '-' -f1 \| cut -d '+' -f1`                                   | `3`                                        |
 
 If you think you need support for other variables, please [open an issue](https://github.com/slsa-framework/slsa-github-generator/issues/new).
+### Multi-platform builds
+Similar to the goreleaser workflow, it's easy to generate binaries for multiple platforms. To accomplish this, we can use the maxtrix functionality of github actions and several config files. In the below code sample, you'll see that we have a `strategy` section which lists the platforms and architectures to build for. These reference a `config-file` property at the bottom which will select the correct config for that platform.
```
Comment (developer-guy) [inline internal/builders/go/README.md:148]: What about giving a link for matrix functionality?
_https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs_

File: internal/builders/go/README.md [hunk 2]
```diff
@@ -144,6 +144,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
...
+
+```
+  build:
+    permissions:
+      id-token: write # To sign the provenance.
+      contents: write # To upload assets to release.
+      actions: read # To read the workflow path.
+    needs: args
```
Comment (developer-guy) [inline internal/builders/go/README.md:156]: I don't think we need to use the `needs` keyword in an example, WDYT?

File: internal/builders/go/README.md [hunk 3]
```diff
@@ -144,6 +144,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
...
+          - windows
+          - darwin
+        arch:
+          - amd64
+          - arm64
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.2.2
+    with:
+      go-version: 1.17
+      # Optional: only needed if using ldflags.
+      evaluated-envs: "COMMIT_DATE:${{needs.args.outputs.commit-date}}, COMMIT:${{needs.args.outputs.commit}}, VERSION:${{needs.args.outputs.version}}, TREE_STATE:${{needs.args.outputs.tree-state}}"
```
Comment (developer-guy) [inline internal/builders/go/README.md:170]: maybe we should remove the `evaluated-envs` keyword for the sake of simplicity, WDYT?

**Review by developer-guy — APPROVED** [review]
not a maintainer but LGTM

File: internal/builders/go/README.md [hunk 4]
```diff
@@ -145,6 +145,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
 
 If you think you need support for other variables, please [open an issue](https://github.com/slsa-framework/slsa-github-generator/issues/new).
 
+### Multi-platform builds
+
+Similar to the goreleaser workflow, it's easy to generate binaries for multiple platforms. To accomplish this, we can use the [maxtrix functionality](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) of github actions and several config files. In the below code sample, you'll see that we have a `strategy` section which lists the platforms and architectures to build for. These reference a `config-file` property at the bottom which will select the correct config for that platform.
```
Comment (laurentsimon) [inline internal/builders/go/README.md:150]: NOTE: goreleaser uses a single config file, so it's more user-friendly. Let's just remove the the `Similar to the goreleaser workflow`, and starts with `It's easy`.

**Review by laurentsimon — APPROVED** [review]
Thanks. Just left a nit. Then let's merge!

File: internal/builders/go/README.md [hunk 5]
```diff
@@ -145,6 +146,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
...
+        os:
+          - linux
+          - windows
+          - darwin
+        arch:
+          - amd64
+          - arm64
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.2.2
+    with:
+      go-version: 1.17
```
Comment (ianlewis) [inline internal/builders/go/README.md:170]: Let's maybe use a more recent Go version ;)
```suggestion
      go-version: 1.19
```

File: internal/builders/go/README.md [hunk 6]
```diff
@@ -145,6 +146,33 @@ The configuration file accepts many of the common fields GoReleaser uses, as you
...
+    strategy:
+      matrix:
+        os:
+          - linux
+          - windows
+          - darwin
+        arch:
+          - amd64
+          - arm64
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.2.2
```
Comment (ianlewis) [inline internal/builders/go/README.md:168]: Let's use the most recent release.
```suggestion
    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.4.0
```

**Review by ianlewis — APPROVED** [review]
I missed the merge but some comments for a potential follow up.

**Pull request merged**